### PR TITLE
VREffect: Support layer bounds

### PR DESCRIPTION
**Pull request by brianchirls**
- Retrieve left and right eye layer bounds on vrdisplaypresentchange event
- Set viewport to layer bounds when rendering stereo

There is a test example here http://output.jsbin.com/fohuqen
The example changes the output resolution every 2 seconds just to show that it works, though this is not how it would be done in practice. This will work in the WebVR Chrome build, though there is currently a [bug](https://github.com/toji/chrome-webvr-issues/issues/79) that causes the view to reset every time the resolution changes. @toji said he's working on it.

This will also work with the polyfill. My example uses code that I will be submitting as a pull request to fix dynamic resolution support there.

### CODE REVIEW

**Comment by mrdoob** [comment]
Uh... What are layer bounds...?

**Comment by toji** [comment]
When calling `VRDisplay.requestPresent` you define a `VRLayer` that supplies the canvas you want to present (`source`) and, optionally, the areas of that canvas that should be displayed in each eye. (The layer bounds.) Those default to the left and right halves of the canvas, respectively, but you can define them to be anything you want. This is useful if you want to render a scene monoscopically (by rendering one view and defining overlapping eye bounds) or if you want to dynamically scale the resolution your rendering at. Since resizing the canvas is a bit expensive it's faster to reset the layer bounds to a smaller area of the canvas and render only to that.

File: examples/js/effects/VREffect.js [hunk 1]
```diff
@@ -116,23 +116,16 @@ THREE.VREffect = function ( renderer, onError ) {
 	var requestFullscreen;
 	var exitFullscreen;
 	var fullscreenElement;
+	var leftBounds = [ 0.0, 0.0, 0.5, 0.5 ];
```
Comment (mrdoob) [inline examples/js/effects/VREffect.js]: Shouldn't this be `[ 0.0, 0.0, 0.5, 1.0 ]`?

File: examples/js/effects/VREffect.js [hunk 2]
```diff
@@ -141,17 +134,35 @@ THREE.VREffect = function ( renderer, onError ) {
 				eyeWidth = eyeParamsL.renderWidth;
 				eyeHeight = eyeParamsL.renderHeight;
 
+				if ( vrDisplay.getLayers ) {
+
+					var layers = vrDisplay.getLayers();
+					if (layers.length) {
+
+						leftBounds = layers[0].leftBounds || [ 0.0, 0.0, 0.5, 1.0 ];
```
Comment (mrdoob) [inline examples/js/effects/VREffect.js]: `leftBounds` and `rightBounds` are already `[ 0.0, 0.0, 0.5, 1.0 ]` and `[ 0.5, 0.0, 0.5, 1.0 ]`.

How about just doing:

```js
if ( layers[ 0 ] ) {
    leftBounds = layers[ 0 ].leftBounds;
    rightBounds = layers[ 0 ].rightBounds;
}
```

File: examples/js/effects/VREffect.js (hunk 1 above)
Comment (brianchirls) [inline examples/js/effects/VREffect.js]: Yes. I will fix.

File: examples/js/effects/VREffect.js (hunk 2 above)
Comment (brianchirls) [inline examples/js/effects/VREffect.js]: I believe the `VRLayer` object returned by `getLayers` can have `leftBounds` or `rightBounds` be `null` if either property was not set. The default value is only represented internally.

@toji can you confirm?

Comment (toji) [inline examples/js/effects/VREffect.js]: Confirmed. The layer object returned is the one you passed in (or a copy of it.) If you pass in `null` for one of the bounds or leave it undefined you get the same thing back out. Willing to hear reasons why this should change if anyone feels strongly about it, but it seems like pretty logical behavior to me.

Comment (brianchirls) [inline examples/js/effects/VREffect.js]: Thanks, @toji. I don't see any reason to change this. The spec is very clear about what the default values are, so if they're not specified you can still know what they are.

Though I do feel the layer object should be a copy of the original so nothing gets screwed up if someone changes the values inside.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**